core/src/main/scala/cats/arrow/FunctionK.scala

```diff
@@ -89,74 +89,14 @@ object FunctionK {
    *
    * Additionally, the type parameters on `f` must not be specified.
    */
-  def lift[F[_], G[_]](f: (F[α] ⇒ G[α]) forSome { type α }): FunctionK[F, G] =
-    macro FunctionKMacros.lift[F, G]
+  def lift[F[_]]: LiftPartiallyApplied[F] = new LiftPartiallyApplied
 
-}
-
-private[arrow] object FunctionKMacros {
```

Comment:
We can't simply remove this code. we need to create two versions of it, one for scala 2.x with the current implementation and another with the new scala3 implementation. Like in this example
https://github.com/typelevel/cats/tree/master/core/src/main/scala-2.13%2B/cats/compat
we created two versions of `cats.compat.SortedSet`